pkg/dashboard/oidc.go

```diff
@@ -0,0 +1,59 @@
+package dashboard
+
+import (
+	"log/slog"
+	"net/http"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/gin-gonic/gin"
+	"github.com/rancher/opni/pkg/storage"
+	"golang.org/x/oauth2"
+)
+
+// AuthDataSource provides a way to obtain data which the auth
+// middleware and proxies need.
+type AuthDataSource interface {
+	StorageBackend() storage.Backend
```

Comment:
does this need to encapsulate all the `storage.Backend` methods? if not, it might be better to have a more granular interface here (for separation of concerns and future proofing)

Reply:
We need multiple interfaces (KeyValueBroker and RoleBindingStore) so to keep the types matched up correctly we need to use the top level interface.